spring-boot-project/spring-boot/src/main/java/org/springframework/boot/DigitalAmount.java

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright 2012-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot;
+
+import java.io.Serializable;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import org.springframework.lang.NonNull;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+
+/**
+ * This class models a quantity or amount of digital information in terms of bytes. This
+ * class is immutable and thread-safe.
+ *
+ * @author Dmytro Nosan
+ * @since 2.1.0
+ * @see DigitalAmountStyle
+ * @see DigitalUnit
+ */
+public final class DigitalAmount implements Comparable<DigitalAmount>, Serializable {
+
+	/**
+	 * amount in terms of bytes.
+	 */
+	private final long amount;
+
+	private DigitalAmount(long amount) {
+		this.amount = amount;
+	}
+
+	/**
+	 * Converts this {@code DigitalAmount} to the bytes.
+	 * @return the total amount of the bytes.
+	 */
+	public long toBytes() {
+		return this.amount;
+	}
+
+	/**
+	 * Converts this {@code DigitalAmount} to the kilobytes.
+	 * @return the total amount of the kilobytes.
+	 */
+	public long toKilobytes() {
+		return DigitalUnit.BYTES.toKilobytes(this.amount);
+	}
+
+	/**
+	 * Converts this {@code DigitalAmount} to the megabytes.
+	 * @return the total amount of the megabytes.
+	 */
+	public long toMegabytes() {
+		return DigitalUnit.BYTES.toMegabytes(this.amount);
+	}
+
+	/**
+	 * Converts this {@code DigitalAmount} to the gigabytes.
+	 * @return the total amount of the gigabytes.
+	 */
+	public long toGigabytes() {
+		return DigitalUnit.BYTES.toGigabytes(this.amount);
+	}
+
+	/**
+	 * Converts this {@code DigitalAmount} to the terabytes.
+	 * @return the total amount of the terabytes.
+	 */
+	public long toTerabytes() {
```

Comment:
All those `to` can go away in benefit of a generic `public long to(Unit unit)` (more on that later).